### DISCUSSION TITLE
Using "sensible interfaces"

### DISCUSSION THREAD

**Discussion by ramsey** [php-identifier/identifier#7] [Ideas]
I received feedback on Twitter[^1] regarding the use of the suffixes `*Interface` (and, to a lesser extent, `*Exception`). The discussion pointed to a post by @mathiasverraes on using "[Sensible Interfaces](https://verraes.net/2013/09/sensible-interfaces/)."

I'm opening this discussion thread here to continue this discussion and iterate on the ideas presented on Twitter.

[^1]: https://twitter.com/_Codito_/status/1576363136986517504

**Comment by MGatner** [comment]
I definitely agree! That Verraes article gets to the heart of the matter: it isn't just a "naming thing", but an implication of design expectations. I appreciate your thoughtfulness about PSR- and legacy- consistency, but let's leave this one behind.

**Comment by drupol** [comment]
I couldn't agree more !

I also usually use sensible interfaces and move them all in their own directory, exemple project: https://github.com/ecphp/cas-lib/tree/master/src

All the interfaces are under the directory `contract`.

PS: using discussions for technical stuff is so much better than Twitter 👍

**Reply by ramsey** [reply]
It looks like you still add the `*Interface` suffixes to your interface names.

**Reply by drupol** [reply]
True, I didn't do it everywhere and it's an issue that I'm planning to fix for version 3.

**Comment by ramsey** [comment]
I've changed the interface names in https://github.com/php-identifier/identifier/commit/147c60dac3304e3919493846cb552d319f9f2116
